src/wallet/rpcwallet.cpp

```diff
@@ -2575,6 +2577,258 @@ UniValue fundrawtransaction(const JSONRPCRequest& request)
+    // (no point in bumping fee if we know that the new tx won't be accepted to the mempool)
+    // This may occur if the user set TotalFee or paytxfee too low, if fallbackfee is too low, or, perhaps,
+    // in a rare situation where the mempool minimum fee increased significantly since the fee estimation just a
+    // moment earlier. In this case, we report an error to the user, who may use totalFee to make an adjustment.
+    CFeeRate minMempoolFeeRate = mempool.GetMinFee(GetArg("-maxmempool", DEFAULT_MAX_MEMPOOL_SIZE) * 1000000);
+    if (nNewFeeRate.GetFeePerK() < minMempoolFeeRate.GetFeePerK()) {
+        strError = strprintf("New fee rate (%s) is too low to get into the mempool (min rate: %s)", FormatMoney(nNewFeeRate.GetFeePerK()), FormatMoney(minMempoolFeeRate.GetFeePerK()));
+        throw JSONRPCError(RPC_MISC_ERROR, strError);
+    }
+
+    CAmount nDelta = nNewFee - nOldFee;
+    if (nDelta <= 0) // it should not be possible to have a negative delta at this point (an attempt to reduce the fee)
+        throw JSONRPCError(RPC_INVALID_PARAMETER, "New fee must be higher than old fee");
+
+    {
+        LOCK(mempool.cs);
+        auto it = mempool.mapTx.find(hash);
+        if (it != mempool.mapTx.end()) {
+
+            // Tx with descendants
+            // --------------------
+            // Idea:  When a tx is bumped, its descendants (if any) are evicted from the mempool.
+            // Policy is that when you replace a tx, the total fees in the mempool cannot go down.
+            // So when you bump a tx that has children, you have to bump your fee by the sum of all of the
+            // descendants, plus the new bumped tx's relay fee.
+            //
+            // If the bumped fee is less than what's required because of children, we fail.  This is a different
+            // situation than bumping the fee to pay the minimum relay fee. Here, we can't be sure whether the user
+            // really wants to pay full price for all of the child transactions.  If so, the user can set payTxFee
+            // and run the command again.
+            //
+            CAmount nFeesWithDescendantsPlusRelay = it->GetModFeesWithDescendants() + ::minRelayTxFee.GetFee(txSize);
+            if (nNewFee < nFeesWithDescendantsPlusRelay) {
+                uint64_t numDescendants = it->GetCountWithDescendants()-1;
+                uint64_t sizeDescendants = it->GetSizeWithDescendants() - it->GetTxSize();
+                std::string strError = strprintf("Insufficent fee due to the child transactions.\n");
+                strError += strprintf("The bumped fee must be at least: %s.\n", FormatMoney(nFeesWithDescendantsPlusRelay));
+                strError += strprintf("Number of child transactions: %u, total size of child transactions: %u\n", numDescendants, sizeDescendants);
+                strError += strprintf("Note that the child transactions would be evicted from the mempool and would not be mined.\n");
```

Comment:
s/would/may probably/

In the future, it may make sense to combine some children transactions (ie, our own) into this one while bumping the fee.